core/templates/components/forms/custom-forms-directives/image-uploader-modal.component.ts

```diff
@@ -0,0 +1,208 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for image uploader modal.
+ */
+
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  Input,
+  ViewChild,
+} from '@angular/core';
+import {SafeResourceUrl} from '@angular/platform-browser';
+import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
+import {ConfirmOrCancelModal} from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import Cropper from 'cropperjs';
+import {SvgSanitizerService} from 'services/svg-sanitizer.service';
+import {WindowDimensionsService} from 'services/contextual/window-dimensions.service';
+require('cropperjs/dist/cropper.min.css');
+
+@Component({
+  selector: 'oppia-image-uploader-modal',
+  templateUrl: './image-uploader-modal.component.html',
+})
+export class ImageUploaderModalComponent extends ConfirmOrCancelModal {
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() allowedImageFormats!: string[];
+  @Input() bgColor!: string;
+  @Input() previewFooter!: string;
+  @Input() previewDescription!: string;
+  @Input() previewTitle!: string;
+  @Input() previewDescriptionBgColor!: string;
+  @Input() imageName!: string;
+  @Input() aspectRatio!: string;
+  @Input() maxImageSizeInKB!: number;
+  @Input() previewImageUrl!: string;
+
+  // 'uploadedImage' will be null if the uploaded svg is invalid or not trusted.
+  uploadedImage: SafeResourceUrl | null = null;
+  thumbnailImageDataUrl: string | null = null;
+  croppedImageDataUrl: string | SafeResourceUrl | null = null;
+  invalidImageWarningIsShown: boolean = false;
+  windowIsNarrow: boolean = false;
+  invalidTagsAndAttributes: {tags: string[]; attrs: string[]} = {
+    tags: [],
+    attrs: [],
+  };
+  dimensions = {height: 0, width: 0};
+  imageType!: string;
+
+  // 'cropper' is initialized before it is to be used, hence we need to do
+  // non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  cropper!: Cropper;
+  @ViewChild('croppableImage') croppableImageRef!: ElementRef;
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    private ngbActiveModal: NgbActiveModal,
+    private windowDimensionService: WindowDimensionsService,
+    private svgSanitizerService: SvgSanitizerService
+  ) {
+    super(ngbActiveModal);
+  }
+
+  private _getAspectRatio(): number {
+    return (
+      parseInt(this.aspectRatio.split(':')[0]) /
+      parseInt(this.aspectRatio.split(':')[1])
+    );
+  }
+
+  initializeCropper(): void {
+    if (this.croppableImageRef) {
```

Comment:
This is one of the code style guidelines I find almost always applicable

If it's simple to do so, make your code as "left" as possible

Instead of 
```
if (a) {
  <logic>
}
```

Write

```
if (!a) {
  return
}
<logic>
```

Here, you can return immediately if croppableImageRef is null

Reply:
Sure I'll follow this. updated the code.